Pieces follow the mouse and lock into the grid more smoothly

Duarpeto opened the pull request:
# Description

Updated the code so that the pieces have better movement when held.
This is closer to what I saw of the real game, and I think it feels better.

## Related issue(s)

Not exactly related to an issue, although it should change #42. I'm not 100% happy with the rotation yet though,  it could be smoother.

## List of changes

- The mouse will always be grabbing the same cell of the piece, which will maintain even when rotating
- When over the grid, the piece will move to its placed position and not move until the mouse moves out of the same grid cell
<br/>

- GameManager.gd - Various changes in piece movement and settling
- GridLogic.gd - Added function that checks if a position is off grid, rather than a piece
- PieceLogic.gd - Changed _input() and related functions to send the clicked cell index to the game manager

Casulono commented on Birthday-2024-Project/Scripts/PieceLogic.gd:
```diff
@@ -252,18 +259,19 @@ func _SetReturnPoint():
-			return true # Mouse is inside one of the piece's cells
-	return false
+			return i # Mouse is inside one of the piece's cells
+	return -1
```
same case as with 0.1, have it as constant value NO_CELL or something that would have similar meaning

Duarpeto commented:
### Update 

- Added _mouse_has_moved() function to GameManager.gd to make code more readable
- Added NO_CELL_CLICKED constant to PieceLogic.gd to make code more readable
- Added logic to rotate functions in PieceLogic.gd to move the piece at the same time as rotating and smoothing said rotation

Duarpeto commented on Birthday-2024-Project/Scripts/GameManager.gd:
```diff
@@ -154,30 +164,35 @@ func _rotate_held_piece():
 	var mouse_distance_moved : float = (mouse_position - previous_mouse_position).length()
-	if mouse_distance_moved > 0.1:
+	var mouse_grid_pos : Vector2i = grid.PositionToGridCoordinate(mouse_position)
+	if (mouse_distance_moved > 0.1 and mouse_grid_pos != previous_mouse_grid_pos) or grid.CheckIfOffGridPos(mouse_position):
```
This specific magic number I think shouldn't be a constant, since as far as I understand it's just a "did the mouse actually move beyond possible floating point errors?" number with no significance. I tried to make it more readable by putting the logic in a separate function with a comment
